torchaudio/functional/filtering.py

```diff
@@ -877,12 +877,20 @@ def lfilter(
 
     input_signal_windows.div_(a_coeffs[0])
     a_coeffs_flipped.div_(a_coeffs[0])
-    for i_sample, o0 in enumerate(input_signal_windows.t()):
-        windowed_output_signal = padded_output_waveform[
-            :, i_sample:i_sample + n_order
-        ]
-        o0.addmv_(windowed_output_signal, a_coeffs_flipped, alpha=-1)
-        padded_output_waveform[:, i_sample + n_order - 1] = o0
+
+    if input_signal_windows.device.type=='cpu' and\
```

Comment:
You can also use the [device object](https://pytorch.org/docs/master/tensor_attributes.html#torch.torch.device) directly for comparisons instead of using the type field.